libs/vkd3d/command.c

```diff
@@ -9988,6 +9988,76 @@ static void STDMETHODCALLTYPE d3d12_command_list_SetPredication(d3d12_command_li
     }
 }
 
+// *PIXEvent* enum values and consts are from PIXEventsCommon.h of winpixeventruntime package
+
+static char* decode_pix_blob(const void* data, size_t size)
+{
+  char* label_str = NULL;
+
+  enum PIXEventType
+  {
+    ePIXEvent_EndEvent = 0x000,
+    ePIXEvent_BeginEvent_VarArgs = 0x001,
+    ePIXEvent_BeginEvent_NoArgs = 0x002,
+    ePIXEvent_SetMarker_VarArgs = 0x007,
+    ePIXEvent_SetMarker_NoArgs = 0x008,
+
+    ePIXEvent_EndEvent_OnContext = 0x010,
+    ePIXEvent_BeginEvent_OnContext_VarArgs = 0x011,
+    ePIXEvent_BeginEvent_OnContext_NoArgs = 0x012,
+    ePIXEvent_SetMarker_OnContext_VarArgs = 0x017,
+    ePIXEvent_SetMarker_OnContext_NoArgs = 0x018,
+  };
+
+  bool isANSI;
+  UINT64 Type;
+
+  size_t StrLength;
+
+  UINT64* data64 = (UINT64*) (data);
+
+  static const UINT64 PIXEventsTypeReadMask = 0x00000000000FFC00;
+  static const UINT64 PIXEventsTypeBitShift = 10;
+
+  static const UINT64 PIXEventsStringIsANSIReadMask = 0x0040000000000000;
+
+  Type = (*data64 & PIXEventsTypeReadMask) >> PIXEventsTypeBitShift;
+
+  if (Type == ePIXEvent_SetMarker_NoArgs)
+    Type = ePIXEvent_BeginEvent_NoArgs;
+
+  if (Type != ePIXEvent_BeginEvent_NoArgs)
+  {
+    WARN("Unexpected/unsupported PIX3Event type, SetMarker_VarArgs and SetMarker_VarArgs can be added on request");
+    return NULL;
+  }
+
+  data64++; //skip Color
+  data64++;
+
+  isANSI = *data64 & PIXEventsStringIsANSIReadMask;
+  
+  data64++;
+
+  if (isANSI)
+  {
+    StrLength = strnlen_s((const char*)data64, SIZE_MAX);
```

Comment:
Just use `strlen`. It's not useful to use `strnlen` here to protect against iterating over the whole address space. :frog:


FTR: The `_s` functions are originally meme MSVC extensions that got erroneously promoted in C11, when they should have just used the ubiquitous POSIX standard ones. If you do need to use `strnlen` again, just use `strnlen`.